WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginWpcomService.java

```diff
@@ -0,0 +1,384 @@
+
+        mIdToken = intent.getStringExtra(ARG_SOCIAL_ID_TOKEN);
+        mService = intent.getStringExtra(ARG_SOCIAL_SERVICE);
+        isSocialLogin = intent.getBooleanExtra(ARG_SOCIAL_LOGIN, false);
+
+        AccountStore.AuthenticatePayload payload = new AccountStore.AuthenticatePayload(email, password);
+        mDispatcher.dispatch(AuthenticationActionBuilder.newAuthenticateAction(payload));
+        AppLog.i(T.NUX, "User tries to log in wpcom. Email: " + email);
+
+        return START_REDELIVER_INTENT;
+    }
+
+    private void handleAuthError(AccountStore.AuthenticationErrorType error, String errorMessage) {
+        if (error != AccountStore.AuthenticationErrorType.NEEDS_2FA) {
+            AnalyticsTracker.track(AnalyticsTracker.Stat.LOGIN_FAILED, error.getClass().getSimpleName(),
+                    error.toString(), errorMessage);
+
+            if (isSocialLogin) {
+                AnalyticsTracker.track(AnalyticsTracker.Stat.LOGIN_SOCIAL_FAILURE, error.getClass().getSimpleName(),
+                        error.toString(), errorMessage);
+            }
+        }
+
+        switch (error) {
+            case INCORRECT_USERNAME_OR_PASSWORD:
+            case NOT_AUTHENTICATED: // NOT_AUTHENTICATED is the generic error from XMLRPC response on first call.
+                setState(LoginPhase.FAILURE_EMAIL_WRONG_PASSWORD);
+                break;
+            case NEEDS_2FA:
+                // login credentials were correct anyway so, offer to save to SmartLock
+                signalCredentialsOK();
+
+                if (isSocialLogin) {
+                    setState(LoginPhase.FAILURE_SOCIAL_2FA);
+                } else {
+                    setState(LoginPhase.FAILURE_2FA);
+                }
+
+                break;
+            case INVALID_REQUEST:
+                // TODO: FluxC: could be specific?
+            default:
+                setState(LoginPhase.FAILURE);
+                AppLog.e(T.NUX, "Server response: " + errorMessage);
+
+                ToastUtils.showToast(this, errorMessage == null ? getString(R.string.error_generic) : errorMessage);
+                break;
+        }
+    }
+
+    protected void startPostLoginServices() {
+        // Get reader tags so they're available as soon as the Reader is accessed - done for
+        // both wp.com and self-hosted (self-hosted = "logged out" reader) - note that this
+        // uses the application context since the activity is finished immediately below
+        ReaderUpdateService.startService(getApplicationContext(), EnumSet.of(ReaderUpdateService
+                .UpdateTask.TAGS));
+
+        // Start Notification service
+        NotificationsUpdateService.startService(getApplicationContext());
+    }
+
+    private void fetchAccount() {
+        setState(LoginPhase.FETCHING_ACCOUNT);
+        mDispatcher.dispatch(AccountActionBuilder.newFetchAccountAction());
+    }
+
+    private EventBus getEventBus() {
```

Comment:
this method looks unnecessary - we can just call `EventBus.getDefault()`

Reply:
Right! I had grand plans for the `getEventBus()` method but when the plans changed I forgot to remove it. Fixed with d4f65a6.